src/main/java/gov/cms/madie/services/CqmMeasureMapper.java

```diff
@@ -0,0 +1,123 @@
+package gov.cms.madie.services;
+
+import gov.cms.madie.models.measure.*;
+import gov.cms.madie.qrda.*;
+import gov.cms.madie.util.ElmDependencyUtil;
+import org.json.JSONObject;
+import org.mapstruct.Mapper;
+import org.mapstruct.Mapping;
+import org.mapstruct.MappingConstants;
+
+import java.time.ZoneId;
+import java.time.format.DateTimeFormatter;
+import java.util.*;
+import java.util.stream.Collectors;
+
+@Mapper(componentModel = MappingConstants.ComponentModel.SPRING)
+public interface CqmMeasureMapper {
+
+  @Mapping(target = "hqmf_set_id", source = "measure.measureSetId")
+  @Mapping(target = "hqmf_version_number", source = "measure.versionId")
+  @Mapping(target = "cms_id", source = "measure.measureSet.cmsId")
+  @Mapping(target = "title", source = "measure.measureName")
+  @Mapping(target = "description", source = "measure.measureMetaData.description")
+  @Mapping(target = "measure_scoring", expression = "java(measure.getScoring().toUpperCase())")
+  @Mapping(target = "main_cql_library", source = "measure.cqlLibraryName")
+  @Mapping(target = "calculate_sdes", source = "measure.testCaseConfiguration.sdeIncluded")
+  @Mapping(target = "calculation_method", expression = "java(getCalculationMethod(measure))")
+  @Mapping(target = "measure_period", expression = "java(getMeasurePeriod(measure))")
+  @Mapping(target = "cql_libraries", expression = "java(getCqlLibraries(measure, elms))")
+  CqmMeasure measureToCqmMeasure(QdmMeasure measure, List<String> elms);
+
+  default String getCalculationMethod(QdmMeasure measure) {
+    return measure.isPatientBasis() ? "PATIENT" : "EPISODE_OF_CARE";
+  }
+
+  default MeasurePeriod getMeasurePeriod(QdmMeasure measure) {
+    JSONObject low = new JSONObject();
+    low.put(
+        "value",
+        measure
+            .getMeasurementPeriodStart()
+            .toInstant()
+            .atZone(ZoneId.systemDefault())
+            .toLocalDateTime()
+            .format(DateTimeFormatter.ofPattern("yyyyMMddHH")));
+
+    JSONObject high = new JSONObject();
+    high.put(
+        "value",
+        measure
+            .getMeasurementPeriodEnd()
+            .toInstant()
+            .atZone(ZoneId.systemDefault())
```

Comment:
need to confirm how this will react with measurement period coming as UTC from UI

Reply:
@chubert-sb - resolving this for purposes of merge, but let's follow up on this with some dedicated testing and make sure it outputs properly in the export